app/models/backend_api_config.rb

```diff
@@ -30,6 +30,12 @@ class BackendApiConfig < ApplicationRecord
     System::Database.oracle? ? common_query.where('path is NOT NULL') : common_query.where.not(path: '')
   end
 
+  scope :accessible, -> do
+    joining { [service, backend_api] }.where.has do
+      (service.state != ::Service::DELETE_STATE) & (backend_api.state != ::BackendApi::DELETED_STATE)
```

Comment:
Just wondering... is it really possible to have a backend api config associated to a deleted backend api?

Reply:
I think it is the safest approach for now :smile: Specially because right now in the API side we allow that.
But as I told you, as I discovered that from the UI we don't allow that, and from the API we do, we definitely should have both sides behaving the same way :joy: 
Let's talk about all this topic more in depth outside here though :+1: